state/watcher/watcher.go

```diff
@@ -103,6 +103,10 @@ type event struct {
 	revno int64
 }
 
+// Period is the delay between each sync.
+// It must not be changed when any watchers are active.
```

Comment:
The race is because a test patches the Period value. This is done outside the watcher and is normally ok, except when race test are run. So this is the minimal change to allow the many tests which rely on reducing the period to run ok, but also make everything thread safe.